content/learn/book/migration-guides/0.6-0.7/_index.md

```diff
@@ -0,0 +1,298 @@
+
+The `QueryEntityError` enum now has a `AliasedMutability` variant, and returns the offending entity.
+
+### [Remove margins.rs](https://github.com/bevyengine/bevy/pull/4284)
+
+The `Margins` type was removed. To migrate, replace every occurrence of `Margins` with `Rect`.
+
+### [Remove face_toward.rs](https://github.com/bevyengine/bevy/pull/4277https://github.com/bevyengine/bevy/pull/4277)
+
+The `FaceToward` trait was removed. To migrate, replace every occurrence of `Mat4::face_toward` to `Mat4::look_at_rh`.
+
+### [`World::entities_mut` is now unsafe](https://github.com/bevyengine/bevy/pull/4093)
+
+```rs
+// 0.6
+world.entities_mut()
+
+// 0.7
+unsafe { world.entities_mut() }
+```
+
+### [Custom vertex attributes](https://github.com/bevyengine/bevy/pull/3959)
+
+Custom vertex attributes are now referenced by a `MeshVertexAttribute` rather than a simple string and `set_attribute` has been renamed to `insert_attribute` better reflect its behavior.
+
+```rs
+// 0.6
+mesh.set_attribute("Vertex_Custom", VertexAttributeValues::Sint32x4(vec![]));
+
+// 0.7
+// Generate your own random identifier here.
+// https://play.rust-lang.org/?gist=cc7e824724ba023e9bff25db35ef1f5e
+pub const ATTRIBUTE_CUSTOM: MeshVertexAttribute =
+    MeshVertexAttribute::new("Custom", 17351772347970238659, VertexFormat::Sint32x4);
+mesh.insert_attribute(
+    ATTRIBUTE_CUSTOM,
+    VertexAttributeValues::Sint32x4(vec![]),
+);
+```
+
+### [Mesh vertex buffer layouts](https://github.com/bevyengine/bevy/pull/3959)
+
+Vertex buffers no longer need to be manually laid out with offset and stride values in a `RenderPipelineDescriptor`.
+
+```rs
+// 0.6
+let vertex_buffer_layout = VertexBufferLayout {
+    array_stride: 20,
+    step_mode: VertexStepMode::Vertex,
+    attributes: vec![
+        VertexAttribute {
+            format: VertexFormat::Float32x3,
+            offset: 0,
+            shader_location: 0,
+        },
+        VertexAttribute {
+            format: VertexFormat::Float32x2,
+            offset: 12,
+            shader_location: 1,
+        },
+    ],
+};
+
+// 0.7
+let mut formats = vec![
+    VertexFormat::Float32x3,
+    VertexFormat::Float32x2,
+];
+let vertex_layout = VertexBufferLayout::from_vertex_formats(VertexStepMode::Vertex, formats);
+```
+
+### [Remove the need for 'IntoSystem::into_system()' when using run criteria piping](https://github.com/bevyengine/bevy/pull/3923)
+
+```rs
+// 0.6
+.with_run_criteria(RunCriteria::pipe(
+    "is_done_label",
+    IntoSystem::into_system(inverse),
+))
+
+// 0.7
+.with_run_criteria(RunCriteria::pipe("is_done_label", inverse))
+```
+
+### [Remove RunSystem](https://github.com/bevyengine/bevy/pull/3817)
+
+You probably should not have been using [`RunSystem`] or [`ParamSystem`], but if you were and you really need it, please make sure to let us know by creating a new discussion over at <https://github.com/bevyengine/bevy/discussions>
+
+<!-- TODO add links to RunSystem and ParamSystem maybe? -->
```

Comment:
```suggestion
You probably should not have been using [`RunSystem`] or [`ParamSystem`], but if you were and you really need it, please make sure to let us know by [creating a new discussion](https://github.com/bevyengine/bevy/discussions).

[`RunSystem`]: https://docs.rs/bevy/0.6.1/bevy/ecs/system/trait.RunSystem.html
[`ParamSystem`]: https://docs.rs/bevy/0.6.1/bevy/ecs/system/struct.ParamSystem.html
```

Pretty sure I agree with adding links here (or removing brackets). Also reworked the discussions link.